packages/ketcher-macromolecules/src/components/macromoleculeProperties/CalculateMacromoleculePropertiesButton.tsx

```diff
@@ -0,0 +1,93 @@
+/****************************************************************************
+ * Copyright 2021 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************/
+
+import { useAppDispatch, useAppSelector } from 'hooks';
+import {
+  selectIsMacromoleculesPropertiesWindowOpened,
+  setMacromoleculesPropertiesWindowVisibility,
+} from 'state/common';
+import styled from '@emotion/styled';
+import { Button } from 'ketcher-react';
+import { blurActiveElement } from 'helpers/canvas';
+import { useRecalculateMacromoleculeProperties } from '../../hooks/useRecalculateMacromoleculeProperties';
+
+const StyledButton = styled(Button)<{ isActive?: boolean }>(
+  ({ theme, isActive }) => ({
+    width: '28px',
+    height: '28px',
+    backgroundColor: isActive
+      ? theme.ketcher.color.button.group.active
+      : 'white',
+    margin: '2px',
+    padding: '0',
+    borderRadius: '4px',
+    outline: 'none',
+
+    ':hover': {
+      backgroundColor: isActive
+        ? theme.ketcher.color.button.group.hover
+        : 'white',
+    },
+
+    ':hover svg': {
+      fill: isActive ? 'white' : theme.ketcher.color.button.group.active,
+    },
+  }),
+);
+
+export const CalculateMacromoleculePropertiesButton = () => {
+  const dispatch = useAppDispatch();
+  const isMacromoleculesPropertiesWindowOpened = useAppSelector(
+    selectIsMacromoleculesPropertiesWindowOpened,
+  );
+  const recalculateMacromoleculeProperties =
+    useRecalculateMacromoleculeProperties();
+
+  const handleClick = async () => {
+    const isMacromoleculesPropertiesWindowOpenedNewState =
+      !isMacromoleculesPropertiesWindowOpened;
+
+    await recalculateMacromoleculeProperties();
+    dispatch(
+      setMacromoleculesPropertiesWindowVisibility(
+        isMacromoleculesPropertiesWindowOpenedNewState,
+      ),
+    );
+
+    blurActiveElement();
+  };
+
+  return (
+    <StyledButton
+      isActive={isMacromoleculesPropertiesWindowOpened}
+      onClick={handleClick}
+    >
+      <svg
```

Comment:
Let's separate it into its own file and add to `iconNameToIcon` map

Reply:
Will do in separate pr